.github/workflows/shared-create-release.yml

```diff
@@ -0,0 +1,216 @@
+            echo 'repository unmodified'
+            exit 1
+          fi
+          git commit -m "generate changelog"
+
+      - uses: actions-rs/toolchain@v1
+        if: ${{inputs.release-target == 'rust' || inputs.release-target == 'wasm'}}
+        with:
+          toolchain: stable
+
+      - uses: actions-rs/cargo@v1
+        if: ${{inputs.release-target == 'rust' || inputs.release-target == 'wasm'}}
+        with:
+          command: install
+          args: cargo-release
+
+      - uses: actions-rs/cargo@v1
+        if: ${{inputs.release-target == 'rust'}}
+        with:
+          command: install
+          args: cargo-edit
+
+      - name: Replace Version In Bindings
+        if: ${{inputs.release-target == 'rust'}}
+        run: |
+          cd bindings/wasm
+          cargo add identity@=${{ inputs.tag-base }} --path=../../identity
+
+      - name: Commit changes
+        if: ${{inputs.release-target == 'rust'}}
+        run: |
+          git add .
+          if [[ $(git diff --stat --staged) == '' ]]; then
+            echo 'repository unmodified'
+            exit 1
+          fi
+          git commit -m "bump version"
+      
+      - name: Run cargo-release
+        if: ${{inputs.release-target == 'rust'}}
+        run: |
+          cargo release --workspace --config ./release.toml --no-tag --no-publish --no-push --no-confirm ${{ inputs.tag-base }} --execute
+
+      - name: Run cargo-release
+        if: ${{inputs.release-target == 'wasm'}}
+        run: |
+          cd bindings/wasm
+          cargo release --no-tag --no-publish --no-push --no-confirm ${{ inputs.tag-base }} --execute
+
+      - uses: actions/setup-node@v2
+        if: ${{inputs.release-target == 'wasm'}}
+        with:
+          node-version: '16'
+
+      - name: Replace Version
+        if: ${{inputs.release-target == 'wasm'}}
+        run: |
+          cd bindings/wasm
+          npm version ${{ inputs.tag-base }}
+
+      - name: Commit npm version bump
+        if: ${{inputs.release-target == 'wasm'}}
+        run: |
+          git add .
+          if [[ $(git diff --stat --staged) == '' ]]; then
+            echo 'repository unmodified'
+            exit 1
+          fi
+          git commit -m "bump version"
+          
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v3
+        with:
+          committer: GitHub <noreply@github.com>
+          author: ${{ github.actor }} <${{ github.actor }}@users.noreply.github.com>
+          branch: release/${{ inputs.tag-prefix }}${{ inputs.tag-base }}
+          delete-branch: true
+          title: 'Release ${{ inputs.tag-prefix }}${{ inputs.tag-base }}'
+          body: |
+            This automatically generated PR contains changes for the `${{ inputs.tag-prefix }}${{ inputs.tag-base }}` version.
+            ${{inputs.pr-body-text}}
+            If you discover any mistakes fix them with commits on this branch. If you want to abort the release simply close the PR.
```

Comment:
TODO: add link to release guide in PR

Reply:
will do via inputs.pr-body-text once document is ready